pkg/controllers/nodeclass/status/ami.go

```diff
@@ -32,7 +32,7 @@ type AMI struct {
 }
 
 func (a *AMI) Reconcile(ctx context.Context, nodeClass *v1beta1.EC2NodeClass) (reconcile.Result, error) {
-	amis, err := a.amiProvider.Get(ctx, nodeClass, &amifamily.Options{})
+	amis, err := a.amiProvider.UpdateAMIs(ctx, nodeClass, &amifamily.Options{})
```

Comment:
What's the reason we think the name should be changed? The function itself isn't actually doing an update, it's doing a retrieval so I think that Get() or List() makes sense